heron/common/src/cpp/network/connection.cpp

```diff
@@ -62,25 +61,26 @@ Connection::~Connection() {
   delete mIncomingPacket;
   {
     for (auto iter = mOutstandingPackets.begin(); iter != mOutstandingPackets.end(); ++iter) {
-      delete iter->first;
+      delete *iter;
```

Comment:
iter points to an element which is a pointer, so *iter is the pointer, delete *iter means deleting the contents pointed by that pointer, it is fine.